src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerSaveImage.groovy

```diff
@@ -35,14 +37,74 @@ class DockerSaveImage extends AbstractDockerRemoteApiTask {
     /**
      * Where to save image.
      */
+
     @OutputFile
     final RegularFileProperty destFile = project.objects.fileProperty()
 
+    /**
+     * Output file containing the image IDs of the saved images.
+     * Defaults to "$buildDir/.docker/$taskpath-imageIds.properties".
+     * If path contains ':' it will be replaced by '_'.
+     *
+     * @since 7.4.0
+     */
+
+    @OutputFile
+    final RegularFileProperty imageIdsFile = project.objects.fileProperty()
+
     DockerSaveImage() {
         useCompression.set(false)
+        String safeTaskPath = path.replaceFirst("^:", "").replaceAll(":", "_")
+        imageIdsFile.set(project.layout.buildDirectory.file(".docker/${safeTaskPath}-imageIds.properties"))
+
         onlyIf {
             images.getOrNull()
         }
+
+        outputs.upToDateWhen {
```

Comment:
This method became quite long. Could you break this up a little into other methods? After that I think we are ready to merge.

Reply:
Yeah, sure.
I stupidly had the code twice anyway for reading and writing.
Hope it suits better now. :-)